sha1_mb/sha1_mb_rand_update_test.c

```diff
@@ -116,15 +117,29 @@ int main(void)
 			ctx = sha1_ctx_mgr_submit(mgr,
 						  &ctxpool[i],
 						  buf_ptr[i], UPDATE_SIZE, HASH_UPDATE);
-
-		// Add jobs while available or finished
-		if ((ctx == NULL) || hash_ctx_complete(ctx)) {
-			i++;
+		if (ctx != NULL && hash_ctx_idle(ctx)) {
+			// Resubmit unfinished job
+			i = (unsigned long)(ctx->user_data);
+			buf_ptr[i] += UPDATE_SIZE;
 			continue;
 		}
-		// Resubmit unfinished job
-		i = (unsigned long)(ctx->user_data);
-		buf_ptr[i] += UPDATE_SIZE;
+		if (ctx) {
+			assert(ctx->error == 0);
+		}
+
+		/* ctx is null or ctx is free
```

Comment:
if ctx is null or context status is complete , it will run below code to find next job

Reply:
@gbtucker , complete status will run the code after line 130